### ISSUE TITLE
Issue with the special characters in the encoded URLs

### ISSUE THREAD

**Issue by habilinour** [MetPX/sarracenia#109]
sr_credentials.get  should decode the special characters in the encoded URLs  before retunring them

**Comment by michelgrenier** [comment]
Yesterday while migrating the GUDFX suite there were 17 senders to migrate... and so we had at least these 17 credentials to deal with ( + 2 polls... etc) .    We had issues with a few of them and luckily we (me, Kurt, Jun, Dominic)  were able to make then work.   The issues were

Question mark ?   in  the password
Slashe  /   in the password

Basically what these characters does is that following the URL parsing rules (urllib.parse.urlparse)
what is after the ?   is a query ...   and I did not investigated how the url was parsed with a / in the password... but it did not work.

In the URL world  a  ?   could be encoded with  %3F  (if I recall correctly ... not sure)
We support space with %20 in file/directory name.  

Encoding a ? with %3F and translating the url,  to me,  just move the problem...
We could literally run into the case where  the string %3F  is a real part of the
 password or maybe the user who knows...

These are rare cases... and probably they will be meet only during the migration
One can think that when everything runs using sarracenia,  sources and clients,
will be advise that a particular set of characters cannot be used in the username and password.

If these restrictions are unacceptable, the sr_credential.py file could be rewritten.
where urllib would not be used anymore... and the credentials.conf file could use
another format not requiring the parsing of the url.   (ex.:   URL  user= password= ... other options)

IMHO

**Comment by petersilva** [comment]
yes ? becomes %3F, and if you want to encode that string literally, you just encode the percent... %25F
I don't see a problem with that, I don't think there are any restrictions based on use of the url-encoding mechanism.   but yeah, decoding them before returning them would make a lot of sense, to be able to use them as-is.  We should make that change.

**Comment by michelgrenier** [comment]
If I well understood ... DI people write the credentials as provided by the source or client  as is  in credentials.conf (same for resolving destination).   In the sr_credentials.py code,  before parsing the credentials'url string   we encode it  (urllib.parse.quote(urlstr))  which turns it into all the % coded stuff,
we parse it as we do now  (and with the encoding there should be no more problem)  and finally when it is time to get the real value of the username  and password  we decode them (urllib.parse.unquote(username) and urllib.parse.unquote(password)....  I guess that yes that would work for everything but spaces  (because we split the string to get the different fields)  ... yup agreed, another thing on Benoit's table.

**Comment by petersilva** [comment]
spaces are %20 ..

**Comment by petersilva** [comment]
I´m wondering if we should just update the documentation (there is an example somewhere) to include the decode in it´s use until we are ready to make the change... not 100% sure the change is actually a good thing.

**Comment by michelgrenier** [comment]
I know … we are supporting it in the filepath…. remember ?
